pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
+		name := CustomName + strings.ToLower(bundlePackage.Name)
+		packages = append(packages, convertBundlePackageToPackage(bundlePackage, name, pc.bundle.APIVersion))
+	}
+	return packages, nil
+}
+
+func (pc *PackageClient) WritePackagesToFile(packages []packagesv1.Package, d string) error {
+	directory := filepath.Join(d, packageLocation)
```

Comment:
I see your viewpoint but also looking at it from another angle, if we create the files in the directory directly, there might be several files (package configurations in this case) that are created. This will create an explosion of files in the directory specified. Maybe I could add a message mentioning the location the files will be generated in.